table/block_based_table_builder.cc

```diff
@@ -76,10 +76,18 @@ FilterBlockBuilder* CreateFilterBlockBuilder(
   } else {
     if (table_opt.partition_filters) {
       assert(p_index_builder != nullptr);
+      // Since after partition cut request from filter builder it takes time
+      // until index builder actully cuts the partition, we take the lower bound
+      // as partition size.
+      uint64_t partition_size =
+          table_opt.metadata_block_size >
+                  (uint64_t)table_opt.block_size_deviation
+              ? table_opt.metadata_block_size - table_opt.block_size_deviation
```

Comment:
block_size_deviation is not a fraction, it is rather in number of bytes +/- the target size.